### ISSUE TITLE
macos-icon-screen-color and macos-icon-ghost-color do not work in themes

### ISSUE THREAD

**Issue by nickkadutskyi** [ghostty-org/ghostty#3215]
I have dark and light themes that provide `macos-icon-ghost-color` and `macos-icon-screen-color` but they do not change icon style on first load or when changing system theme.

Icon always appears like this which is a completely wrong color:
![Image](https://github.com/user-attachments/assets/52fc08b5-5e27-48d2-858c-556aa5ed66fe)

[dark theme](https://github.com/nickkadutskyi/nixos-config/blob/ff3f8635d76b27e981db6ee0ca954b7dcf8d5089/users/nick/ghostty/themes/jb-dark#L31)
```
macos-icon = custom-style
macos-icon-ghost-color = BCBEC4
macos-icon-screen-color = 1E1F22
```
and [light theme](https://github.com/nickkadutskyi/nixos-config/blob/ff3f8635d76b27e981db6ee0ca954b7dcf8d5089/users/nick/ghostty/themes/jb-light#L31)
```
macos-icon = custom-style
macos-icon-ghost-color = 000000
macos-icon-screen-color = FFFFFF
```

I've tried to set color to red and green and still it won't apply. 
It only applies when I put `macos-icon-screen-color` and `macos-icon-ghost-color` into `config` file while `macos-icon = custom-style` works when provided by theme file.

**Comment by qwerasd205** [comment]
Duplicate of #3076

**Comment by nickkadutskyi** [comment]
@qwerasd205 I've seen that but it's about the wrong color. My issue that the color won't work at all if I provide it via theme.
E.g. setting this in theme file doesn't work while in config file sets screen color to green:
```
macos-icon-ghost-color = BCBEC4
macos-icon-screen-color = 00FF00
```

**Comment by mitchellh** [comment]
Going to reopen to investigate. It should work!

**Comment by qwerasd205** [comment]
The icon will only ever be generated on startup. Works as expected for me. I just tested putting this at the bottom of my theme config:
```
macos-icon = custom-style
macos-icon-ghost-color = #BCBEC4
macos-icon-screen-color = #00FF00
```
then saving and restarting Ghostty (completely) and it worked just fine. The green is much darker than would possibly be desirable, but that's that same issue.

**Comment by qwerasd205** [comment]
The appearance you of the icon in your screenshot is what you get for your light configuration. I do think we really had ought to support hot reloading of the colorized icon stuff since changing with the system theme would be fun, but as far as I can tell there is absolutely no problem loading these from config files specified by `theme`.

**Comment by nickkadutskyi** [comment]
I did another test:
1. Set system theme to light mode
2. Set in light theme file this: 
```
macos-icon-ghost-color = #BCBEC4
macos-icon-screen-color = #00FF00
```
3. Set in config file this:
```
macos-icon = custom-style
```
4. Remove anything related to `macos-icon` in dark theme file
5. Restart Ghostty
6. I get green screen color. 
7. Set system theme to dark mode
8. Restart Ghostty
9. I get green screen color

It seems like when I start Ghostty it always applies light theme first thus setting the icon and after that it applies dark theme (after checking actual system theme?) but at this point it doesn't change icon colors.


Here is another test where toggling a system theme while keeping Ghostty running changes the icon:
1. Set system theme to light mode
2. Set in light theme file this: 
```
macos-icon = custom-style
macos-icon-ghost-color = #BCBEC4
macos-icon-screen-color = #00FF00
```
3. Remove anything related to `macos-icon` in config file
4. Set in dark theme file this:
```
macos-icon = official
```
5. Restart Ghostty
6. I get green screen color
7. Toggle system theme
8. Icon color changed to how official icon look like
9. Toggle system theme again
10. Icon color stays on the official icon

**Comment by qwerasd205** [comment]
Yes, it will go from colorized -> official when hot reloaded but not the other way around. If Ghostty is applying light mode first that is a problem though.

**Comment by qwerasd205** [comment]
In that case the real problem of this issue is that the light theme config is applied when it shouldn't be, which could theoretically be a problem with other things too- anything that is queried once after launch and then never again.

**Comment by mitchellh** [comment]
> If Ghostty is applying light mode first that is a problem though.

This shouldn't be a problem, just slightly wasteful (and you may see a shimmer I suppose of the wrong theme). We load the config on app init but we don't have an API to tell Ghostty what the system theme is until _after_ initialized. This shouldn't be a root cause for this bug and I think we chalk this up to being unrelated.

**Comment by flaviuvlaicu** [comment]
Is it possible to change the color scheme of the icon even when the terminal is closed? Have a different scheme for when open and closed?

**Comment by appleton** [comment]
I'm getting some weird behaviour where I seem to be able to set the background to certain colours (e.g. the green suggested above) but when I pick `#F6F8FA` I get a black background instead.

E.g. this config:

```
macos-icon = custom-style
macos-icon-frame = aluminum
macos-icon-ghost-color = #BCBEC4
macos-icon-screen-color = #00FF00
```
Yields this icon:

<img width="113" alt="Image" src="https://github.com/user-attachments/assets/ac4024bd-7031-4e97-abca-4dffdb7d930f" />

But this config:

```
macos-icon = custom-style
macos-icon-frame = aluminum
macos-icon-ghost-color = #9F5EBC
macos-icon-screen-color = #F7F8FA
```

Yields this icon:

<img width="119" alt="Image" src="https://github.com/user-attachments/assets/a3b35868-baea-4a2e-982e-ac684765f7fb" />

I would expect the background to be basically white in the second example.

**Comment by qwerasd205** [comment]
@appleton that's unrelated to this issue and is already covered by #3076
